legacy/projects/state-mach-regex/extractlinks/extract_links.py

```diff
@@ -2,36 +2,51 @@
 import sys
 import csv
 if __name__ == '__main__':
-  # Exit if command line args entered incorrectly
-  if len(sys.argv) != 2:
-    print("usage: extract_links.py [input_file]")
-    sys.exit(0)
+    # Exit if command line args entered incorrectly
+    if len(sys.argv) != 2:
+        print("usage: extract_links.py [input_file]")
+        sys.exit(0)
 
 # Filename is 2nd command line arg
 filename = sys.argv[1]
 
-# TODO Read HTML file
+# Read HTML file
 
+with open(filename) as file:
+    filedata = file.read()
+    file.close()
 
-# TODO Set up regex
+# Set up regex
 
+regex = r'https?:\/\/[0-9A-z-]+.?\w+.[A-z]{2,3}\/?[0-9A-z\/=@#%&;_.?-]+'
```

Comment:
That regex has a small flaw that will add an extra match for `https://secure` which is an invalid link. Quicu fix for that is to remove the dot after first bracket check like this:

`https?:\/\/[0-9A-z-]+?\w+.[A-z]{2,3}\/?[0-9A-z\/=@#%&;_.?-]+`

Changing this line of code fixes the issue and properly matches all the links 👍